src/runtime_src/core/common/api/xrt_elf.cpp

```diff
@@ -137,6 +149,11 @@ elf(std::istream& stream)
   : detail::pimpl<elf_impl>{std::make_shared<elf_impl>(stream)}
 {}
 
+elf::
+elf(const void *data, size_t size)
+    : detail::pimpl<elf_impl>{std::make_shared<elf_impl>(data, size)}
+{}
```

Comment:
Actually, just do it here and remove the code in elf_impl.
```suggestion
elf::
elf(const void *data, size_t size)
    : elf{boost::interprocess::ibufferstream{static_cast<const_char*>(data), size}}
{}
```

Reply:
This does not compile, because the elf(std::istream &strm) constructor takes a non const reference? If we make it const then we cannot read() from it :-(

```
/home/sonals/git/XRT/src/runtime_src/core/common/api/xrt_elf.cpp: In constructor ‘xrt::elf::elf(const void*, size_t)’:
/home/sonals/git/XRT/src/runtime_src/core/common/api/xrt_elf.cpp:153:33: error: cannot bind non-const lvalue reference of type ‘std::istream&’ {aka ‘std::basic_istream<char>&’} to an rvalue of type ‘std::basic_istream<char>’
  153 |     :  elf{boost::interprocess::ibufferstream{static_cast<const char*>(data), size}}
      |                                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/sonals/git/XRT/src/runtime_src/core/common/api/xrt_elf.cpp:147:19: note:   initializing argument 1 of ‘xrt::elf::elf(std::istream&)’
  147 | elf(std::istream& stream)
      |     ~~~~~~~~~~~~~~^~~~~~
```

Reply:
Oh well, makes sense.